yw_website/yw_website/apps/website/utils.py

```diff
@@ -89,3 +90,24 @@ def get_direct_descendants(program_block_id, program_blocks):
         new_child.direct_descendants = get_direct_descendants(new_child.id, program_blocks)
         descendants.append(new_child)
     return descendants
+
+def get_ports(parents):
+    for block in parents:
+        for port in Port.objects.filter(on_program_block_id=block.id):
```

Comment:
just out of curiosity, is this behaving in the same way that was causing a problem last time we filtered on a program_block_id?

Reply:
I believe that this should be working the right way, as on_program_block_id references the id of the program block